doc/arch/adr-002-decide-how-to-display-more-facilities.md

```diff
@@ -0,0 +1,249 @@
+in the web app to work in PL/pgSQL. Moreover, each time we added a new filter
+or search option to the web application, we'd have to write a version of the
+same query in PL/pgSQL for the tiler.
+
+##### Security
+
+Martin's `query_params` appear to be passed in to the database as strings, which
+opens a security hole. While we could create a PostGIS role or user with a
+limited, readonly set of permissions to use solely for the Martin instance,
+doing so requires taking on some additional risk and complexity.
+
+Likewise, adding PostGIS-based security just for the tiler may also compel us to
+have to figure out how to duplicate features like API key authentication or
+facilities-data request logging -- which we've already written once in Django.
+
+##### Unfamiliarity
+
+We don't have any experience running Martin in production. We've also got
+limited experience using Rust, the language in which Martin is written.
+Together this means a Martin-based tile server may be difficult to operate and
+debug.
+
+### Adding a Vector Tile Endpoint to the Existing Django Web App
+
+Adding a vector tile endpoint to the existing web app seemed like a promising
+approach, since it would enable trying out `ST_AsMVT` while reusing the app's
+database connection and Django's querysets for filtering. In this approach we
+would add a `/tile/{layer}/{z}/{x}/{y}/` endpoint to the Django application,
+then update the client to make tile requests there rather than rendering the
+`/facilities` GeoJSON response as Leaflet markers.
+
+#### Pros
+
+##### Provides Access to the Existing Django Queryset Apparatus
+
+While using Martin (or a similar solution) would compel writing new versions of
+the facilities queries in PL/pgSQL, placing a vector tile endpoint in Django
+lets us reuse some of the existing query code and also provides access to
+Django models and querysets. Likewise, we would not have to write new code for
+new filter and search options in two different languages.
+
+##### Already Has a Secure Database Connection
+
+The Django web application already has a secure database connection, so we
+would not have to create a solution for securing Martin or another PostGIS-backed
+tile server.
+
+##### Enables Switching from `ST_AsMVT` to Another Python Vector Tile Option
+
+There remains some question about the viability of using `ST_AsMVT`. If it turns
+out that this is not a performant solution, having the tile endpoint in Django
+makes it possible to drop out of using `ST_AsMVT` altogether and to switch to
+an alternate Python library for generating vector tiles.
+
+##### Doesn't Require Creating & Deploying a Different Service
+
+Adding Martin or another vector tile server would increase the number of
+different kinds of services running as part of the OAR, which adds to the
+application's complexity. Keeping the tile endpoint in Django does not require
+adding a new service.
+
+##### Allows Scaling by Increasing the Number of App Instances
```

Comment:
I don't think this warrants a change to this ADR nor the decision you've arrived at, but it may be valuable to point out that this may not be a pro (it may not be a con either, but is certainly a consequence). I understood the implication of this statement to be that you _lose_ the ability to scale the the tiling and the API endpoints separately, they must be done in tandem. You also lose the ability to provision the API infrastructure independently of the tiler. For example, if the tiling endpoints end up requiring higher memory allocations, you will be obligated to provision all API instances with the extra resources, and will take that hit if you need to scale on the API-only dimension.